src/main/java/com/epam/aidial/core/controller/LimitController.java

```diff
@@ -18,18 +21,51 @@ public LimitController(Proxy proxy, ProxyContext context) {
         this.context = context;
     }
 
-    public Future<?> getLimits(String deploymentName) {
-        proxy.getRateLimiter().getLimitStats(deploymentName, context).onSuccess(limitStats -> {
-            if (limitStats == null) {
-                context.respond(HttpStatus.NOT_FOUND);
-            } else {
-                context.respond(HttpStatus.OK, limitStats);
+    public Future<?> getLimits(String deploymentId) {
+        Deployment deployment = context.getConfig().selectDeployment(deploymentId);
+
+        Future<Deployment> deploymentFuture;
+        if (deployment != null) {
+            if (!DeploymentController.hasAccess(context, deployment)) {
+                log.error("LimitController. Forbidden deployment {}. Key: {}. User sub: {}", deploymentId, context.getProject(), context.getUserSub());
+                return context.respond(HttpStatus.FORBIDDEN, "Forbidden deployment: " + deploymentId);
```

Comment:
Can we throw exceptions instead of responding to have clean flow? E.g.
Future.compose().compose().compose().onSuccess(e-> respond).onFailure(e-> respond)

Reply:
fixed